Server/models/Check.js

```diff
@@ -31,4 +31,27 @@ const CheckSchema = mongoose.Schema(
   }
 );
 
+/**
+ * When a check is created, we should update the stauts of the associated monitor
+ * if it has changed.  Monitor starts with default status: true
+ */
+
+CheckSchema.pre("save", async function (next) {
+  const monitor = await mongoose.model("Monitor").findById(this.monitorId);
+  if (monitor.status !== this.status) {
+    if (monitor.status === true && this.status === false) {
+      // TODO issue alert
+      console.log("Monitor went down");
+    }
+
+    if (monitor.status === false && this.status === true) {
+      // TODO issue alert
+      console.log("Monitor went up");
+    }
+    await monitor.save();
+  }
+
+  next;
```

Comment:
Is this a typo? Shouldn't it be `next()` instead?

Reply:
Yeah. its a function:  `next()`

Reply:
Good catch, typing too fast :P